packages/xstate-react/src/createProvider.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import { createContext, useContext } from 'react';
+import { useInterpret } from './useInterpret';
+import { useActor } from './useActor';
+import { useSelector } from './useSelector';
+import { ActorRefFrom, AnyStateMachine, EmittedFrom } from 'xstate';
+import { EventFrom } from '@xstate/fsm';
+
+export function createProvider<TMachine extends AnyStateMachine>(
+  machine: TMachine
+): {
+  (props: { children: React.ReactNode }): JSX.Element;
```

Comment:
I'm curious! why ```React.ReactElement<any,any>``` is better option here that ```JSX.Element```

Reply:
They are pretty much the same - `JSX.Element` is actually defined like this:
```ts
declare global {
    namespace JSX {
        interface Element extends React.ReactElement<any, any> { }
    }
}
```

The main difference for me is that... we have stronger guarantees that we'll resolve the correct thing. Support for JSX in TS is weirdly implemented, so I just make sure that we don't try to resolve to a global `JSX` since that could be somehow tampered with. 

It also matched how the builtin `React.FC` (and more) are typed - so I want to follow their suite.